core/model/src/main/java/it/unibz/inf/ontop/model/term/functionsymbol/db/impl/DefaultDBStrStartsWithFunctionSymbol.java

```diff
@@ -43,6 +42,32 @@ public String getNativeDBString(ImmutableList<? extends ImmutableTerm> terms,
                         secondTerm));
     }
 
+    @Override
+    protected ImmutableTerm buildTermAfterEvaluation(ImmutableList<ImmutableTerm> newTerms, TermFactory termFactory, VariableNullability variableNullability) {
+        if (newTerms.get(1) instanceof DBConstant) {
+            String prefixString = ((DBConstant) newTerms.get(1)).getValue();
+
+            if (newTerms.get(0) instanceof DBConstant) {
+                DBConstant firstConstant = (DBConstant) newTerms.get(0);
+                return termFactory.getDBBooleanConstant(firstConstant.getValue().startsWith(prefixString));
+            }
+
+            if (newTerms.get(0) instanceof ImmutableFunctionalTerm) {
+                ImmutableFunctionalTerm firstNonFunctionalTerm = (ImmutableFunctionalTerm) newTerms.get(0);
```

Comment:
Please rename the variable (it is actually functional) into `firstFunctionalTerm`